src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/FixAllSpanMappingService/AbstractFixAllSpanMappingService.cs

```diff
@@ -0,0 +1,91 @@
+            if (decl == null)
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
+
+            if (fixAllScope == FixAllScope.ContainingMember)
+            {
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                    .SetItem(document, ImmutableArray.Create(decl.FullSpan));
+            }
+            else
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = semanticModel.GetDeclaredSymbol(decl, cancellationToken);
+                if (symbol?.DeclaringSyntaxReferences.Length > 1)
+                {
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                    _ = PooledDictionary<Document, ImmutableArray<TextSpan>>.GetInstance(out var builder);
+                    foreach (var syntaxRef in symbol.DeclaringSyntaxReferences)
+                    {
+                        var documentForLocation = document.Project.GetDocument(syntaxRef.SyntaxTree);
+                        if (documentForLocation == null)
+                            continue;
+                        var root = await syntaxRef.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+                        var partialDeclSpan = syntaxFacts.GetContainingTypeDeclaration(root, syntaxRef.Span.Start)!.FullSpan;
+                        builder.MultiAdd(documentForLocation, partialDeclSpan);
+                    }
+
+                    return builder.ToImmutableDictionary();
+                }
+                else
+                {
+                    return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                        .SetItem(document, ImmutableArray.Create(decl.FullSpan));
+                }
+            }
+        }
+
+        private static async Task<SyntaxNode?> GetContainingMemberOrTypeDeclarationAsync(
+            Document document,
+            FixAllScope fixAllScope,
+            TextSpan triggerSpan,
+            CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            var startContainer = fixAllScope == FixAllScope.ContainingMember
+                ? syntaxFacts.GetContainingMemberDeclaration(root, triggerSpan.Start)
+                : syntaxFacts.GetContainingTypeDeclaration(root, triggerSpan.Start);
+
+            if (fixAllScope == FixAllScope.ContainingMember && !syntaxFacts.IsMethodLevelMember(startContainer))
+                return null;
+
+            if (triggerSpan.IsEmpty || startContainer == null)
```

Comment:
1. can you split this heck, it's pretty confusing.  especially "if start is null, return start".